Document deprecation of opencage functions

jessesadler opened the pull request:
* Closes #72
* Bring back opencage_key() as done in Dan's opencage_key branch.
* Move opencage_format() to deprecated.R
* Add warnings to deprecated functions.
* Change documentation to not deprecated functions.
* Change name of test file for deprecated functions and use supressWarnings() in tests that use deprecated functions.

Commit:
Document deprecation of opencage functions

* Closes #72
* Bring back opencage_key() as done in Dan's opencage_key branch.
* Move opencage_format() to deprecated.R
* Add warnings to deprecated functions.
* Change documentation to not deprecated functions.
* Change name of test file for deprecated functions and use supressWarnings() in tests that use deprecated functions.

dpprdan commented on tests/testthat/test-oc_check_query.R:
```diff
@@ -98,11 +98,12 @@ test_that("oc_check_query checks language", {
-      language = "fr-NOTACOUNTRYCODE"
-    ),
+    suppressWarnings(
+      opencage_forward(
```
Nice catch! However, this ought to test oc_check_query IMO.

dpprdan commented on tests/testthat/test-deprecated.R:
```diff
@@ -4,33 +4,35 @@ test_that("opencage_forward/opencage_reverse return what they should.", {
   skip_if_offline()
 
-  results <- opencage_forward(placename = "Sarzeau")
+  results <- suppressWarnings(opencage_forward(placename = "Sarzeau"))
```
Instead of `suppressWarnings` you could also test for a warning with `expect_warning` like [this](https://github.com/dpprdan/opencage/blob/411edf89f3093fdcc71b3ada9266462841d580fd/tests/testthat/test-opencage_forward.R#L6). This depends on how we want to deal with the deprecated warnings in general though (see my main comment below).

Commit:
Use expect_warning for deprecated functions

Commit:
Fix merge conflicts

Merge branch 'devel' into deprecated

# Conflicts:
#	tests/testthat/test-deprecated.R
